wp-content/plugins/core/src/Theme/Config/Font_Sizes.php

```diff
@@ -0,0 +1,65 @@
+<?php
+declare( strict_types=1 );
+
+namespace Tribe\Project\Theme\Config;
+
+class Font_Sizes {
+	/**
+	 * @var array
+	 */
+	private $sizes;
+
+	/**
+	 * Colors constructor.
+	 *
+	 * @param array $sizes An array of size definitions.
+	 *                     Keys should be unique identifiers for the sizes.
+	 *                     Values should contain an array of:
+	 *                     - size  - int - the font size, in pixels
+	 *                     - label - string - the translated label for the size
+	 */
+	public function __construct( array $sizes ) {
+		$this->sizes = $sizes;
+	}
+
+	/**
+	 * Get a list of size options.
+	 *
+	 * @return array
+	 */
+	public function get_sizes(): array {
+		return $this->sizes;
+	}
+
+	/**
+	 * Return a subset of the size list.
+	 *
+	 * @param string[] $size_keys
+	 *
+	 * @return self
+	 */
+	public function get_subset( array $size_keys = [] ): self {
+		return new self( array_intersect_key( $this->sizes, array_flip( $size_keys ) ) );
+	}
+
+	/**
+	 * Formats the sizes as `[ 'name' => <label>, 'slug' => <key>, 'size' => <size> ]`
+	 *
+	 * This is the format expected by the block editor.
+	 *
+	 * @return array
+	 * @see https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-font-sizes
+	 */
+	public function format_for_blocks(): array {
```

Comment:
I may not understand completely, but I don't think we want to use an actual font-size control since our sizes are generally different per viewport size; and, we apply sizes based on css classes (or specific html elements inside a css class).

Reply:
These are the sizes that a user can select in the block editor for, e.g., paragraph blocks. It functions by applying a class to the paragraph, so the theme could handle it differently depending on viewport, etc. The font sizes we give are for admin display purposes only.

These are entirely optional. If we don't want to give users the ability to shrink/embiggen text, we can just return an empty array here.